library/ssl_tls12_server.c

```diff
@@ -735,18 +737,21 @@ static int ssl_pick_cert( mbedtls_ssl_context *ssl,
         MBEDTLS_SSL_DEBUG_CRT( 3, "candidate certificate chain, certificate",
                           cur->cert );
 
+        int key_type_mismatch = 0;
```

Comment:
Non-blocker: making this `key_type_matches` would make the code easier to read. I would have inverted the condition in the original code too.